logging/src/logging.cc

```diff
@@ -34,38 +33,61 @@ namespace leatherman { namespace logging {
     static bool g_colorize = false;
     static bool g_error_logged = false;
 
+    class color_writer : public sinks::basic_sink_backend<sinks::synchronized_feeding>
+    {
+     public:
+        color_writer(ostream *dst);
+        void consume(boost::log::record_view const& rec);
+     private:
+        ostream *_dst;
+    };
+
+    color_writer::color_writer(ostream *dst) : _dst(dst) {}
+
+    void color_writer::consume(boost::log::record_view const& rec)
+    {
+        auto level = boost::log::extract<log_level>("Severity", rec);
+        auto line_num = boost::log::extract<int>("LineNum", rec);
+        auto name_space = boost::log::extract<string>("Namespace", rec);
+        auto timestamp = boost::log::extract<boost::posix_time::ptime>("TimeStamp", rec);
+        auto message = rec[expr::smessage];
+
+        (*_dst) << boost::gregorian::to_iso_extended_string(timestamp->date());
+        (*_dst) << " " << boost::posix_time::to_simple_string(timestamp->time_of_day());
+        (*_dst) << " " << left << setfill(' ') << setw(5) << level << " " << *name_space;
+        if (*line_num > 0) {
```

Comment:
I see the check for 0 here, but I think it'd make sense not to attach the attribute when 0 rather than doing so and just not writing the message if it is.  I guess I could go either way on this, the more I think about it.